Do a full build with Mingw on CI

Early versions of the Mingw environment didn't have libraries setup correctly for linking, hence why we only compiled to object files and never tried to link them. It may have been because libraries were not installed (only header files were), or the library paths were never setup correctly. The newer Mingw build environments should have that corrected.

----

To account for the deficiencies in the old Mingw environment, a custom `makefile` target (`intermediate`) was created that avoided the linking step. This was used in the CircleCI config for the Mingw environment:
> `- run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" intermediate`

We should switch this to the full `build` job that other environments use:
> `- build`

----

Currently it is possible to do a full build and link of OPHD in the updated Mingw environment. However there is a build failure for the `testLibOPHD` project: **undefined reference to `SDL_main'**

Details:
> x86_64-w64-mingw32-g++ .build/Debug_Linux_testLibOPHD/Intermediate/MapOffset.test.o .build/Debug_Linux_testLibOPHD/Intermediate/RandomNumberGenerator.test.o .build/Debug_Linux_libOPHD/libOPHD.a nas2d-core/lib/libnas2d.a  -L/usr/local/x86_64-w64-mingw32/lib -lmingw32 -mwindows **-lSDL2main** -lSDL2 -lgtest **-lgtest_main** -lgmock **-lgmock_main** -lpthread  -lSDL2 -lSDL2_image -lSDL2_mixer -lSDL2_ttf -lglew32 -lopengl32 -o .build/Debug_Linux_testLibOPHD/testLibOPHD
> /usr/bin/x86_64-w64-mingw32-ld: /usr/local/x86_64-w64-mingw32/lib/libSDL2main.a(SDL_windows_main.c.obj):SDL_windows_ma:(.text+0x152): **undefined reference to `SDL_main'**
> collect2: error: ld returned 1 exit status
> make: *** [makefile:223: .build/Debug_Linux_testLibOPHD/testLibOPHD] Error 1
> make: Target 'testLibOPHD' not remade because of errors.

The following link flags should be noted:
- `-lSDL2main`
- `-lgtest_main`
- `-lgmock_main`

For unit test projects we should be using either `gmock_main` (if any Google Mock features are used), or `gtest_main` (if we rely only on Google Test features). We should remove all but one of the libraries offering a `main` function.

On a related note, we might also want to re-order the libraries being passed to the linker. Related:
- https://github.com/lairworks/nas2d-core/pull/1186

----

It's questionable if we want to link any SDL libraries for this unit test project. This unit test project does rely on `libOPHD`, which relies on `NAS2D`, which relies on `SDL`. In that sense we potentially rely on SDL. However, the features of NAS2D that we rely on for `libOPHD` shouldn't be the ones that rely on SDL. For instance, we use features such as `Point`, `Vector`, and `Rectangle` almost universally, along with a number of other features that get used a bit less often. These features are fairly abstract, and don't rely on external libraries. These are the features we need in `libOPHD`, and they are the ones that don't rely on SDL.

This might be an indication that NAS2D should be split. There are features that are useful to core game logic in places like `libOPHD`, whereas the features relying on SDL would be more relevant to `libControl` and `appOPHD`. NAS2D is a separate project though, so splitting it would be out of scope for this issue.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -51,10 +51,12 @@ jobs:
   build-linux-mingw:
     docker:
       - image: outpostuniverse/nas2d-mingw:1.11
+    environment:
+      LDFLAGS_EXTRA: -L/usr/local/x86_64-w64-mingw32/lib
     steps:
       - checkout
       - run: git submodule update --init nas2d-core/
-      - run: make --keep-going --jobs 16 CXXFLAGS_EXTRA="-Werror" intermediate
+      - build
 
 workflows:
   build:

--- a/makefile
+++ b/makefile
@@ -154,7 +154,7 @@ testLibControls_CPPFLAGS := $(CPPFLAGS) -I./
 testLibControls_LDLIBS := -lgmock_main -lgmock -lgtest -lpthread $(LDLIBS)
 
 testLibControls_PROJECT_FLAGS := $(testLibControls_CPPFLAGS) $(CXXFLAGS)
-testLibControls_PROJECT_LINKFLAGS = $(LDFLAGS) $(testLibControls_LDLIBS)
+testLibControls_PROJECT_LINKFLAGS = $(LDFLAGS_EXTRA) $(testLibControls_LDLIBS)
 
 .PHONY: testLibControls
 testLibControls: $(testLibControls_OUTPUT)

